cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java

```diff
@@ -271,22 +269,15 @@ public List<LocationGroup> getLocationGroups(String officeId, String locCategory
         }
 
         if (officeId != null) {
-            if (CWMS.equalsIgnoreCase(officeId)) {
-                connectBy = onStep.where(alcg.CAT_DB_OFFICE_ID.eq(CWMS)
-                        .and(alcg.GRP_DB_OFFICE_ID.eq(CWMS))
-                        .and(condition)
-                );
-            } else {
-                connectBy = onStep.where(alcg.CAT_DB_OFFICE_ID.in(CWMS, officeId)
-                        .and(alcg.GRP_DB_OFFICE_ID.in(CWMS, officeId))
-                        .and(alga.DB_OFFICE_ID.isNull().or(alga.DB_OFFICE_ID.eq(officeId)))
-                        .and(condition)
-                );
-            }
+            connectBy = onStep.where(DSL.upper(alcg.CAT_DB_OFFICE_ID).in(officeId.toUpperCase())
```

Comment:
You're searching for location groups with the given office id, not categories with that office id